--- a/Makefile	2024-02-23 00:12:53.804891093 +0300
+++ b/Makefile	2024-02-23 00:10:21.807597190 +0300
@@ -120,7 +120,7 @@
 # ----------
 
 # Defines the operating system and architecture
-override CFLAGS += -DYQ2OSTYPE=\"$(YQ2_OSTYPE)\" -DYQ2ARCH=\"$(YQ2_ARCH)\"
+override CFLAGS += -DYQ2_OSTYPE=\"$(YQ2_OSTYPE)\" -DYQ2_ARCH=\"$(YQ2_ARCH)\"
 
 # ----------
 
--- a/CMakeLists.txt	2024-02-25 13:39:48.225732782 +0300
+++ b/CMakeLists.txt	2024-02-25 13:39:34.577464324 +0300
@@ -19,13 +19,13 @@
 string(REPLACE "-O3" "-O2" CMAKE_C_FLAGS_RELEASE "${CMAKE_C_FLAGS_RELEASE}")
 
 # Operating system
-add_definitions(-DYQ2OSTYPE="${CMAKE_SYSTEM_NAME}")
+add_definitions(-DYQ2_OSTYPE="${CMAKE_SYSTEM_NAME}")
 
 # Architecture string
-string(REGEX REPLACE "amd64" "x86_64" ARCH "${CMAKE_SYSTEM_PROCESSOR}")
-string(REGEX REPLACE "i.86" "i386" ARCH "${ARCH}")
-string(REGEX REPLACE "^arm.*" "arm" ARCH "${ARCH}")
-add_definitions(-DYQ2ARCH="${ARCH}")
+string(REGEX REPLACE "amd64" "x86_64" YQ2_ARCH "${CMAKE_SYSTEM_PROCESSOR}")
+string(REGEX REPLACE "i.86" "i386" YQ2_ARCH "${YQ2_ARCH}")
+string(REGEX REPLACE "^arm.*" "arm" YQ2_ARCH "${YQ2_ARCH}")
+add_definitions(-DYQ2_ARCH="${YQ2_ARCH}")
 
 # Linker Flags
 if(${CMAKE_SYSTEM_NAME} MATCHES "Windows"
